util/utils.q

```diff
@@ -0,0 +1,605 @@
+  }
+
+// @private
+// @kind function
+// @category utilitiesUtility
+// @desc Find columns of certain types
+// @param tab {table} Data in tabular format
+// @param char {char[]} Type of column to find  
+// @return {symbol[]} Columns containing the type being searched 
+i.findCols:{[tab;char]
+  metaTab:0!meta tab;
+  metaTab[`c]where metaTab[`t]in char
+  }
+
+// @private
+// @kind function
+// @category utilitiesUtility
+// @desc Checks if object is of a specified type
+i.isInstance:.p.import[`builtins][`:isinstance;<]
+
+// @private
+// @kind function
+// @category utilitiesUtility
+// @desc Python datetime module
+i.dateTime:.p.import`datetime
+
+// @private
+// @kind function
+// @category utilitiesUtility
+// @desc Python pandas dataframe module
+i.pandasDF:.p.import[`pandas]`:DataFrame
+
+// @private
+// @kind function
+// @category utilitiesUtility
+// @desc Check that the length of the endog and another parameter
+//   are equal 
+// @param endog {float[]} The endogenous variable
+// @param param {number[][]|number[]} A parameter to compare the length of
+// @param paramName {string} The name of the parameter
+// @returns {::|err} Return an error if they aren't equal
+i.checkLen:{[endog;param;paramName]
+  if[not count[endog]=count param;
+    '"The length of the endog variable and ",paramName," must be equal"
+    ]
+  }
+
+// Metric utility functions
+
+// @private
+// @kind function
+// @category metricUtility
+// @desc Exclude collinear points 
+// @param x {number[]} X coordinate of true positives and false negatives
+// @param y {number[]} Y coorfinate of true positives and false negatives
+// @returns {number[]} any colinear points are excluded
+i.curvePts:{[x;y]
+  (x;y)@\:where(1b,2_differ deltas[y]%deltas x),1b
+  }
+
+// @private
+// @kind function
+// @category metricUtility
+// @desc Calculate the area under an ROC cirve
+// @param x {number[]} X coordinate of true positives and false negatives
+// @param y {number[]} Y coorfinate of true positives and false negatives
+// @returns {number[]} Area under the curve
+i.auc:{[x;y]
+  sum 1_deltas[x]*y-.5*deltas y
+  }
+
+// @private
+// @kind function
+// @category metricUtility
+// @desc Calculate the correlation of a matrix
+// @param matrix {number[]} A sample from a distribution
+// @returns {number[]} The covariance matrix
+i.corrMatrix:{[matrix]
+  devMatrix:dev each matrix;
+  covMatrix[matrix]%devMatrix*/:devMatrix
+  }
+
+// Preproc utility functions
+
+// @private
+// @kind function
+// @category preprocessingUtility
+// @desc Drop any constant numeric values
+// @param data {dictionary} Numerical data
+// @return {dictionary} All keys with zero variance are removed
+i.dropConstant.num:{[num]
+  (where 0=0^var each num)_num
+  }
+
+// @private
+// @kind function
+// @category preprocessingUtility
+// @desc Drop any constant values that aren't numeric
+// @param data {dictionary} Non-numerical data
+// @return {dictionary} All keys with zero variance are removed
```

Comment:
The description indicating that other column drop is related to zero variance doesn't hold in this instance.